pkg/actors/health/healthunit.go

```diff
@@ -0,0 +1,26 @@
+//go:build unit
+// +build unit
+
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package health
+
+import "k8s.io/utils/clock"
+
+// WithClock sets a custom clock (for mocking time).
+func WithClock(clock clock.WithTicker) Option {
```

Comment:
Let's keep this without the build tag. The overhead is basically none. We had to recently un-do a similar thing in dapr/kit as it made writing unit tests for other packages harder (https://github.com/dapr/kit/pull/62)

Reply:
No, this we should not pollute implementation code with test code. The `unit` build tag is a standard test tag used throughout the Dapr project where any testing code is able to make use of the `WithClock` option.

> We had to recently un-do a similar thing in dapr/kit as it made writing unit tests for other packages harder

Why was this is the case? Surely the same argument can be made for interface mocks as well.

Reply:
In the linked PR is says?

>  There are use cases where having those methods available even outside of a unit test is helpful, such as when the objects are instantiated with a clock that could be mocked in the unit test for the parent method.

Why would you want to use a mocked clock outside of a unit test?